### DISCUSSION TITLE
How to determine host IP address from inside the VM?

### DISCUSSION THREAD

**Discussion by jandubois** [lima-vm/lima#1600] [Q&A]
Now, I know the host IP is always `192.168.5.2`, so I can just hard-code it.

But I would prefer to dynamically determine it from the VM configuration. With qemu I would fetch it from the default route:

```console
$ lima ip route | grep 'default via'
default via 192.168.5.2 dev eth0 proto dhcp src 192.168.5.15 metric 100
```

But when using VZ with gvisor-tap, it uses the SlirpDNS as the gateway address:

```console
$ lima ip route | grep 'default via'
default via 192.168.5.3 dev eth0 proto dhcp src 192.168.5.1 metric 100
```

So how can I determine the actual host address to call services from the host?

@balajiv113 Any suggestions?

**Comment by jandubois** [comment]
I just realized that I already hard-code `192.168.5.15` for the guest address, so this question is just idle curiosity if this is all as-expected. I'm going to hard-code the host address as well. 😄

**Reply by balajiv113** [reply]
Logically in route there shouldn't be any difference. Looks like some bug there. Will check and update you shortly.

PS: Just noticed the bug the guest ip 192.168.5.15 is not valid for vz driver

**Reply by jandubois** [reply]
Is there a way to get VZ to use `192.168.5.15` for the guest IP? I actually care more about this than being able to parse the default route for the gateway address, as the fixed IP is the same for QEMU and VZ.

There is a comment about VZ using the DNS address as the gateway address, but I don't understand why this is the case: https://github.com/lima-vm/lima/blob/fedacd268cd72fdb3e2d9b25f3dc0f6baaa8bff1/pkg/networks/usernet/gvproxy.go#L45

**Reply by balajiv113** [reply]
We should be able to use 192.168.5.15 for vz under default network. Need to do some changes 

> DNS address as the gateway address

The confusion is due to how gvisor-tap-vsock supports slirp and we using our qemu variable for the same

**Reply by balajiv113** [reply]
Fixed the IP https://github.com/lima-vm/lima/pull/1612

I will also change the DNS and Gateway mismatch as part of https://github.com/lima-vm/lima/issues/1551
